src/Domain/ElementLogicalCoordinates.hpp

```diff
@@ -0,0 +1,51 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <unordered_map>
+#include <vector>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+class DataVector;
+template <size_t Dim>
+struct BlockLogicalCoordHolder;
+template <size_t VolumeDim>
+class ElementId;
+
+/// \ingroup ComputationalDomainGroup
+///
+/// Holds element logical coordinates of an arbitrary set of points on
+/// a single `Element`.  The arbitrary set of points is assumed to be
+/// a subset of a larger set of points spanning multiple `Element`s,
+/// and this class holds `offsets` that index into that larger set of
+/// points.
+///
+/// \details `offsets.size()` is the same as
+/// `element_logical_coords.number_of_grid_points()`
+template <size_t Dim>
+struct ElementLogicalCoordHolder {
```

Comment:
I don't understand what the point of this class is what so ever. The documentation doesn't even begin to tell me what `offsets` is.  What is `element_logical_coords` and how can it have a `number_of_grid_points()` function?

Is this mapping back into the block coordinates really necessary? It seems to me that once you know the Element and the coordinates in that element you're done. Why do you need to map things back?

Reply:
And of course this PR only addresses the functionality of determining where to interpolate, and not the interpolation, or assembling the interpolated data...

Reply:
Also it was discussed that a future improvement in cases where the target points smoothly change (or there is a time-dependent mapping), passing in the previous block_ids would allow those blocks to be "is_contained" first before falling back to the loop over blocks

Reply:
Okay, I think this summary is useful to add to the Doxygen comment here.

I was understanding the offset as telling you which block logical coordinates the point was associated with. I think this is the same as the inertial coordinate it is associated with and it wasn't at all clear to me that knowing the block logical coordinate was useful. However, knowing how the point corresponds to the initial inertial coordinate I can understand.

Yes, I saw the discussion on Slack.